bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/discovery/FreeboxOsDiscoveryService.java

```diff
@@ -106,50 +114,70 @@ protected void startScan() {
                 List<LanHost> lanHosts = new ArrayList<>(thingHandler.getManager(LanBrowserManager.class).getHosts()
                         .stream().filter(LanHost::reachable).toList());
 
-                discoverServer(bridgeUID);
-                discoverPhone(bridgeUID);
-                discoverPlugs(bridgeUID);
-                discoverRepeater(bridgeUID, lanHosts);
-                discoverPlayer(bridgeUID, lanHosts);
-                discoverVM(bridgeUID, lanHosts);
-                discoverHome(bridgeUID);
+                discoverServer(bridgeUID, lowLogLevel);
+                discoverPhone(bridgeUID, lowLogLevel);
+                discoverPlugs(bridgeUID, lowLogLevel);
+                discoverRepeater(bridgeUID, lanHosts, lowLogLevel);
+                discoverPlayer(bridgeUID, lanHosts, lowLogLevel);
+                if (hasVm) {
+                    discoverVM(bridgeUID, lanHosts, lowLogLevel);
+                }
+                if (hasHomeAutomation) {
+                    discoverHome(bridgeUID, lowLogLevel);
+                }
                 if (thingHandler.getConfiguration().discoverNetDevice) {
-                    discoverHosts(bridgeUID, lanHosts);
+                    discoverHosts(bridgeUID, lanHosts, lowLogLevel);
                 }
             } catch (FreeboxException e) {
-                logger.warn("Error while requesting data for things discovery: {}", e.getMessage());
+                if (lowLogLevel) {
+                    logger.debug("Error while requesting data for things discovery: {}", e.getMessage());
+                } else {
+                    logger.warn("Error while requesting data for things discovery: {}", e.getMessage());
+                }
             }
         }
     }
 
-    private void discoverHome(ThingUID bridgeUID) {
+    private void discoverHome(ThingUID bridgeUID, boolean lowLogLevel) {
         NodeConfigurationBuilder builder = NodeConfigurationBuilder.getInstance();
         try {
             thingHandler.getManager(HomeManager.class).getHomeNodes().forEach(
                     node -> builder.configure(bridgeUID, node).ifPresent(result -> thingDiscovered(result.build())));
         } catch (FreeboxException e) {
-            logger.warn("Error discovering Home: {}", e.getMessage());
+            if (lowLogLevel) {
+                logger.debug("Error discovering Home: {}", e.getMessage());
+            } else {
+                logger.warn("Error discovering Home: {}", e.getMessage());
+            }
```

Comment:
This duplicated logging logic seems a bit cumbersome and hard to maintain. Perhaps one first step could be to extract this logic into a dedicated method, something like:
```java
private void logException(String message, Exception e, boolean lowLogLevel) {
    if (lowLogLevel) {
        logger.debug(message, e.getMessage());
    } else {
        logger.warn(message, e.getMessage());
    }
}
```

This would remove the duplication in each `catch` block.